lib/sdn_epc/forwarder.ex

```diff
@@ -0,0 +1,55 @@
+defmodule SdnEpc.Forwarder do
+  use GenServer
+
+  # Client API
+
+  def start_link do
+    res = GenServer.start_link(__MODULE__, :ok, name: __MODULE__)
+  end
+
+  def subscribe_messages_from_switch(datapatch_id, type) do
+    GenServer.cast __MODULE__,
+    {:handle_switch_msg, datapatch_id: datapatch_id, type: type}
+  end
+
+  def open_ofp_channel(switch_id, ip, port, version) do
+    GenServer.call __MODULE__,
+    {:open_of_channel, %{switch_id: switch_id, ip: ip,
+                         port: port,version: version}}
+  end
+
+  def send_msg_to_switch(datapatch_id, msg) do
+    GenServer.cast __MODULE__,
+    {:send_msg_to_switch, datapatch_id: datapatch_id, msg: msg}
+  end
+
+  def send_msg_to_controller(switch_id, msg) do
+    GenServer.cast __MODULE__,
+    {:send_msg_to_controller, switch_id: switch_id, msg: msg}
+  end
+
+  # Server Callbacks
+
+  def init(:ok) do
+    {:ok, %{}}
+  end
+
+  def handle_call({:open_of_channel, args}, _from, state) do
+    ip_tuple = args[:ip]
+    |> String.split(".")
```

Comment:
Apart from that there's `iex(11)> :inet.parse_address('127.0.0.1')
{:ok, {127, 0, 0, 1}}`

Note the single quotes ' ' (erlang string)